python/ray/serve/_private/metrics_utils.py

```diff
@@ -1,138 +1,69 @@
+import asyncio
 import bisect
 import logging
-import math
-import threading
-import time
+from asyncio import sleep as async_sleep
 from collections import defaultdict
 from dataclasses import dataclass, field
-from typing import Any, Callable, DefaultDict, Dict, List, Optional, Union
+from typing import Callable, DefaultDict, Dict, List, Optional
 
-import ray
 from ray.serve._private.constants import SERVE_LOGGER_NAME
 
 logger = logging.getLogger(SERVE_LOGGER_NAME)
 
 
-class _MetricTask:
-    def __init__(self, task_func, interval_s, callback_func):
-        """
-        Args:
-            task_func: a callable that MetricsPusher will try to call in
-                each loop. It should take no arguments, and return a
-                kwargs dictionary to be used to call the callback_func.
-            interval_s: the interval of each task_func is supposed to be called.
-            callback_func: callback function is called when task_func is done, and
-                the result of task_func is passed to callback_func as the first
-                argument, and the timestamp of the call is passed as the second
-                argument.
-        """
-        self.task_func: Callable[[], Dict[str]] = task_func
-        self.interval_s: float = interval_s
-        self.callback_func: Callable[[Any, float]] = callback_func
-        self.last_ref: Optional[ray.ObjectRef] = None
-        self.last_call_succeeded_time: Optional[float] = time.time()
-
-
 class MetricsPusher:
     """
-    Metrics pusher is a background thread that run the registered tasks in a loop.
+    Starts an asyncio task that repeatedly runs registered tasks.
```

Comment:
There's no need to structure it this way anymore -- the standard way to structure this is to have each task independently `asyncio.sleep` and run the function on its own cadence.